src/librustc_yk_sections/emit_tir.rs

```diff
@@ -321,12 +326,32 @@ impl<'tcx> ToPack<ykpack::Statement> for (&ConvCx<'_, 'tcx, '_>, BasicBlock, &St
             StatementKind::Assign(ref place, ref rval) => {
                 let lhs = (*ccx, place).to_pack();
                 let rhs = (*ccx, &**rval).to_pack();
-                if let ykpack::Place::Local(tvar) = lhs {
+                if let ykpack::Place::Base(ykpack::PlaceBase::Local(tvar)) = lhs {
                     ccx.push_def_site(*bb, tvar);
                 }
                 ykpack::Statement::Assign(lhs, rhs)
             },
-            _ => ykpack::Statement::Unimplemented,
+            StatementKind::SetDiscriminant{ref place, ref variant_index} => {
+                ykpack::Statement::SetDiscriminant{
+                    place: (*ccx, place).to_pack(),
+                    variant_index: variant_index.as_u32(),
+                }
+            },
+            // StorageLive/Dead and fake reads are not useful the tracer. Ignore them.
+            StatementKind::StorageLive(..)
+            | StatementKind::StorageDead(..)
+            | StatementKind::FakeRead(..) => ykpack::Statement::Nop,
+            StatementKind::InlineAsm{..} =>
+                ykpack::Statement::Unimplemented(ykpack::UnimplementedStatement::InlineAsm),
+            // Retagging (for stacked borrows).
+            // https://www.ralfj.de/blog/2018/08/07/stacked-borrows.html
+            StatementKind::Retag(..) =>
+                ykpack::Statement::Unimplemented(ykpack::UnimplementedStatement::Retag),
+            // Type ascription.
+            // https://github.com/rust-lang/rfcs/pull/803
+            StatementKind::AscribeUserType(..) =>
```

Comment:
This and `Retag` are no-ops: https://github.com/rust-lang/rust/blob/d00d639c54ae49a35c0c215cd8161af8c7d2e8ee/src/librustc_codegen_ssa/mir/statement.rs#L97-L100

Reply:
Thanks! They can map to our `Nop` instruction then!